src/sampling/fibonacci.jl

```diff
@@ -0,0 +1,65 @@
+"""
+	FibonacciSampling(n)
+
+Generate samples on a sphere using the Fibonacci lattice method.
+The parameter `n` specifies the number of points to generate.
+In the regular Fibonacci lattice method, the number ϕ is the golden ratio ((1 + √5)/2),
+but different numbers can be used, preferably irrational.
+
+## Example
+
+Sample a sphere with 1000 points:
+
+```julia
+sample(Sphere((0,0,0), 1), FibonacciSampling(100))
+
+# sample using π instead of the golden ration
+sample(Box((0,0),(1,1)), FibonacciSampling(100,π))
+"""
+struct FibonacciSampling <: SamplingMethod
+  n::Int
+  ϕ::Real
+  function FibonacciSampling(n::Int, ϕ::Real=(1 + √5) / 2)
+    if n ≤ 0
+      throw(ArgumentError("Size must be positive"))
+    end
+    new(n, ϕ)
+  end
+end
+
+function sample(::AbstractRNG, geom::Geometry, method::FibonacciSampling)
+  # sample points in square
+  i = 0:(method.n - 1)
+  x, y = mod.(i / method.ϕ, 1), i / method.n
+
+  # get points for each geometry
+  ps = Iterators.map(zip(x, y)) do (x, y)
+    Point(_fibonaccipointstogeom(geom, x, y))
+  end
+
+  # get transformations to turn canonical geometry into current geometry
+  transforms = _fibonaccigeomtransforms(geom)
+
+  Iterators.map(ps) do p
+    p |> transforms
+  end
+end
+
+# these functions map points generated by the Fibonacci lattice method to the corresponding points in different geometries.
+# each function takes a geometry type and coordinates (x, y) and returns the transformed coordinates in the specified geometry.
+_fibonaccipointstogeom(geom::Box{𝔼{2}}, x, y) = @. (x - 0.5, y - 0.5) |> lentype(geom)
```

Comment:
Is this assuming a Box with side 1? One way to properly scale the points is to find a reparametrization of (x,y) in terms of (u, v) coordinates. All geometries are parametrized as in disk(u, v), box(u, v), ... with (u,v) in [0,1]^2. If you call box(0, 0) you get the lower left corner, and box(1, 1) you get the upper right corner.

Reply:
The algorithm described in that Observable example samples from a 1x1 square centered at [0.5,0.5]. Thus, the `_fibonaccipointstogeom` for Box is simply moving the sampled points to the [0,0] center. Once this is done, the `_fibonaccigeomtransforms` actually scales the point to the actual box geometry, by applying translation, scalings and so on.

The code works for any Box, every a transformed box with affine transformations applied to it.

Reply:
Just to reiterate, the code transforms the points to a "canonical geometry" (e.g. a canonical box, a canonical sphere, a canonical disk), and then transforms the points to the actual geometry (e.g. a box with affine transformations, a translated and scaled sphere, a translated and rotated disk).

![image](https://github.com/user-attachments/assets/76b23bb4-f789-4cdb-beb1-d4c4ceb28369)
![image](https://github.com/user-attachments/assets/046b5d8e-55fe-4897-bcca-ffc8f7a43fe4)